gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/base.py.j2

```diff
@@ -88,19 +91,16 @@ class {{ service.name }}Transport(abc.ABC):
         if credentials_file is not None:
             credentials, _ = auth.load_credentials_from_file(
                                 credentials_file,
-                                scopes=scopes,
+                                scopes=self._scopes,
                                 quota_project_id=quota_project_id
                             )
 
         elif credentials is None:
-            credentials, _ = auth.default(scopes=scopes, quota_project_id=quota_project_id)
+            credentials, _ = auth.default(scopes=self._scopes, quota_project_id=quota_project_id)
 
         # Save the credentials.
         self._credentials = credentials
 
-        # Lifted into its own function so it can be stubbed out during tests.
-        self._prep_wrapped_messages(client_info)
-
```

Comment:
why moved `self._prep_wrapped_messages(client_info)` to both grpc and grpc_asyncio?

Reply:
[`self._prep_wrapped_messages`](https://github.com/googleapis/python-automl/blob/bab596b3b0fc04581602dd5817b08f876db846fe/google/cloud/automl_v1beta1/services/prediction_service/transports/base.py#L106-L115) creates a dict of the the method callables. The [`grpc` method callables](https://github.com/googleapis/python-automl/blob/bab596b3b0fc04581602dd5817b08f876db846fe/google/cloud/automl_v1beta1/services/prediction_service/transports/grpc.py#L300-L306) need `self.grpc_channel` to exist.

With the change in this PR the base transport needs to set `self._host`, `self._scopes`, and `self._credential` _before_ the gRPC channel can be created. As a result I moved the `self._prep_wrapepd_messages` call to after channel creation in `grpc` and `grpc_asyncio`. 

Suggestions on cleaner ways to do this much appreciated. :)

Reply:
I see. I guess maybe we can add a comment here in grpc and grpc_asyncio to remind us to keep the order. Thank you for simplifying the logic :)